src/SAWScript/Position.hs

```diff
@@ -30,12 +30,49 @@ import qualified What4.FunctionName as W4
 
 -- Pos ------------------------------------------------------------------------
 
+-- Type inference info, to be used to interpret the positions of
+-- inferred types.
+--
+-- InfFresh means that the term (or pattern) at the given position
+--    caused us to generate a fresh type variable, e.g. the element
+--    type of "[]".
+-- InfTerm means that the term (or pattern) at the given position
+--    prompted us to choose the accompanying type; e.g. "[]" has type
+--    List.
+-- InfContext means that the usage of the term at the given position
+--    prompted us to choose the accompanying type; e.g. in "f x" f has
+--    function type.
```

Comment:
It might be worth mentioning here that `Inference` deliberately does _not_ have an `Ord` instance, as it is meant to be compared using `compareInfQuality` instead.

Reply:
That comparison isn't suitable for e.g. instantiating a map. It's more that there's no 3rd instance because there's no need for one. I added a different comment to clarify.

It's possible that compareInfQuality should get replaced with an infQuality function that returns int such that then one can write that comparison as e.g. "infQuality a < infQuality b". I suspect it would be more idiomatic that way. But I don't really like it that way though as it creates a leaky abstraction.

Reply:
I added a different comment.